CDCgov/prime-reportstream
# Implement interface for error and warning messages

## Context
When there are error and/or warning during CSV uploading, each one is listed in the UI. This can become verbose and unfriendly to the end user. The solution is to consolidate these messages as shown in [these mocks](https://www.figma.com/file/ctpYEoOBWGkUnk9cmXzVp2/ReportStream-Submitter-Front-End?node-id=980%3A3216)

Tech specs proposal detailed here: https://app.zenhub.com/workspaces/prime-data-hub-5ff4833beb3e08001a4cacae/issues/cdcgov/prime-reportstream/1966

## Changes
This PR is the first step towards error/warning message consolidation, creating and implementing an interface for each csv error type. With these objections, the errors and warnings can then be summarized before sending to the client. Specifically this PR:
- Introduces a `ResponseMessage` interface that `ResultDetail` uses instead of a `String` for the message
- Creates data classes for each error type that implement `ResponseMessage`
- Creates a generic `ResponseMsgType` all HL7 serialization warnings/errors since these do not need consolidation

Note: this PR does not change the response data back to the client. The UI is still sent down a list of strings.

## Next Steps
- Consolidate warnings/errors and send client a summarized data set
- Display the warnings/errors in a table as shown in the Figma mocks 

## Test Steps:
1. Create an upload csv with known problems (for example, using the test-waters-data.csv, make the patientAge column blank)
2. Upload csv in the csv uploader
3. Verify that the errors/warnings have not changed (this PR should have no changes to the UI)

![image](https://user-images.githubusercontent.com/87096393/134243668-07cc0e01-5f98-4449-83e6-1df3efb51298.png)


## Checklist

### Testing
